pds4indextools/pds4_create_xml_index.py

```diff
@@ -198,271 +701,781 @@ def update_nillable_elements_from_xsd_file(xsd_file, nillable_elements_info):
-
-    parser.add_argument('--sort-by', type=str, nargs='+',
-                        help='Sort resulting index file by one or more columns')
-
-    parser.add_argument('--clean-header-field-names', action='store_true',
-                        help='Replace the ":" and "/" in the column headers with '
-                             'alternative (legal friendly) characters')
-    parser.add_argument('--extra-file-info', type=str, nargs='+',
-                        choices=['LID', 'filename', 'filepath',
-                                 'bundle_lid', 'bundle'],
-                        help='Add additional columns to the final index file. Choose '
-                             'from the following: "LID", "filename", "filepath", '
-                             '"bundle_lid", and "bundle". If using multiple, separate '
-                             'with spaces.')
-    parser.add_argument('--config-file', type=str,
-                        help='Read a user-specified configuration file.. File must be a '
-                             '.ini file.')
-
-    args = parser.parse_args()
+        clean_headers(df)
+
+    if args.fixed_width:
+        padded_df = pad_column_values_and_headers(df)
+        padded_df.to_csv(output_csv_path, index=False, na_rep='NaN')
+
+    else:
+        df.to_csv(output_csv_path, index=False, na_rep='NaN')
+
+
+def find_base_attribute(xsd_tree, target_name, new_namespaces):
+    """
+    Finds the base attribute of a target element in an XML schema.
+
+    This function searches for the base type of a given target element in the provided
+    XML schema tree. It follows nested type definitions to return the final meaningful
+    base type, such as `pds:ASCII_NonNegative_Integer`.
+
+    Args:
+        xsd_tree (etree._Element): The XML schema tree.
+        target_name (str): The name of the target element to search for.
+
+    Returns:
+        str: The base type of the target element if found, otherwise None.
+
+    Raises:
+        etree.XPathEvalError: If there is an error during XPath evaluation.
+    """
+
+    # Register namespaces
+    namespaces = {
+        'xs': 'http://www.w3.org/2001/XMLSchema',
+        'pds': 'http://pds.nasa.gov/pds4/pds/v1'
+    }
+    namespaces.update(new_namespaces)
+
+    def follow_base_type(base_type):
+        """
+        Recursively follows the base type definitions to find the final base type.
+
+        Args:
+            base_type (str): The initial base type to follow.
+
+        Returns:
+            str: The final base type.
+        """
+        while True:
+            if 'ASCII' in base_type or 'UTF8' in base_type:
+                return base_type
+
+            next_query = (
+                f".//xs:simpleType[@name='{base_type.split(':')[-1]}']"
+                f"//xs:restriction/@base"
+            )
+            try:
+                next_result = xsd_tree.xpath(next_query, namespaces=namespaces)
+            except Exception:
```

Comment:
"Naked" exceptions are generally a bad idea. Do you know which exceptions you want to catch here?